controller/proxy-injector/webhook.go

```diff
@@ -74,9 +72,64 @@ func Inject(
 		}
 		ownerKind = strings.ToLower(ownerRef.Kind)
 	}
+
+	configLabels := configToPrometheusLabels(resourceConfig)
 	proxyInjectionAdmissionRequests.With(admissionRequestLabels(ownerKind, request.Namespace, report.InjectAnnotationAt, configLabels)).Inc()
 
-	if injectable, reasons := report.Injectable(); !injectable {
+	// If the pod's namespace has the opaque ports annotation but the pod does
+	// not, then it should be added to the pod template metadata.
+	opaquePorts, opaquePortsOk := resourceConfig.AnnotateOpaquePorts()
```

Comment:
If both the namespace and the workload have the opaque-ports annotation, the latter will override the former. The case could be made to have the values be merged instead. But what if we just issue an event clarifying that the namespace value was overridden?

Reply:
I think we should keep the behavior of the workload's annotation being the one that is used. Merging could be a little surprising, and it get's more complicated when the annotation is deleted from one of the locations.

I'm not sure I see this as the workload overriding the namespace—more that the namespace is providing a fallback value if the workload does not have it already. Maybe a log line in each block could help in this situation? I'm thinking in the pod annotation block, there would be a `log.Debugf("using %s's %s annotation value", conf.pod.meta.Name, k8s.ProxyOpaquePortsAnnotation)`

Reply:
@alpeb log lines added

Reply:
:+1: